### PR TITLE
Feature/cheapest banana price

### PR DESCRIPTION
**Pull request by kareem-m-alaraj**

### CODE REVIEW

File: src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    private lateinit var theMostExpensiveCity_cairo: CityEntity
+
+    @BeforeEach
+    fun setup(){
+        MockKAnnotations.init(this)
+    }
+
+    @Test
+    fun should_ReturnSortedListByCheapestBananaPrices_When_TheListIsNotEmptyAndPricesIsNotZeroOrInMinus() {
+        // given
```
Comment (ahmdghannam) [inline src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt]: write more information about given like the playlist provides

File: src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        every { theMostExpensiveCity_cairo.cityName } returns "Cairo"
+        every { theMostExpensiveCity_cairo.fruitAndVegetablesPrices.banana1kg } returns 25f
+        resultList.add(theMostExpensiveCity_cairo)
+
+        every { theMostCheapestCity_baghdad.cityName } returns "Baghdad"
+        every { theMostCheapestCity_baghdad.fruitAndVegetablesPrices.banana1kg } returns 15f
+        resultList.add(theMostCheapestCity_baghdad)
+
+        every { dataSource.getAllCitiesData() }returns resultList
+        // when
```
Comment (ahmdghannam) [inline src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt]: and here write what happens after when

File: src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        val actual = GetCitiesHasTheCheapestBananaPricesInteractor(dataSource).execute(theAveragePriceCity_gaza,theMostExpensiveCity_cairo,theMostCheapestCity_baghdad)
+
+        // then
```
Comment (ahmdghannam) [inline src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt]: and here

**Review by ahmdghannam — CHANGES_REQUESTED** [review]

File: src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt (hunk 1 above)
Comment (kareem-m-alaraj) [reply src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt]: done.

File: src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt (hunk 2 above)
Comment (kareem-m-alaraj) [reply src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt]: done.

File: src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt (hunk 3 above)
Comment (kareem-m-alaraj) [reply src/test/kotlin/interactor/GetCitiesHasTheCheapestBananaPricesTest.kt]: done.

**Pull request closed without merging**